packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel.tsx

```diff
@@ -1384,74 +1408,89 @@ export default function RenderPanel({ className }: RenderPanelProps) {
 
       const parent = appDom.getParent(dom, draggedNode);
 
-      const parentChildren = parent ? appDom.getChildNodes(dom, parent).children : [];
-      const totalLayoutColumnSizes = parentChildren.reduce(
-        (acc, child) => acc + (nodesInfo[child.id]?.rect?.width || 0),
-        0,
-      );
-
       const resizePreviewRect = resizePreviewElement?.getBoundingClientRect();
 
       if (draggedNodeRect && resizePreviewRect) {
-        const normalizeColumnSize = (size: number) =>
-          Math.max(0, size * parentChildren.length) / totalLayoutColumnSizes;
+        if (draggedEdge === RECTANGLE_EDGE_LEFT || draggedEdge === RECTANGLE_EDGE_RIGHT) {
```

Comment:
I'd suggest to brainstorm how this code could be made flatter, it's cyclomatic complexity seems to keep growing. In addition we should also consider how to split this file into smaller chuncks as +1.5k LoC doesn't seem to be very scalable IMO

Reply:
With the risk of turning this in a religious debate, I don't like using LoC as a metric for maintainability. I've seen people trash projects by splitting code that semantically belongs together purely for the sake of reducing LoC per file.
I have no problem with 5k LoC in a file, nor do I have problems with having 100 files in a folder. As long as code is semantically grouped together, and the principle of single responsibility is respected.

Reply:
I'm not using LoC as a metric, but at certain number I'd say it signals that it could be split and that it might be doing too many things. Without spending too much time in the code my initial impressions that the code inside this component should be mostly responsible for rendering panel. This most recent addition takes care of resizing - so that might be considered as another responsibility, then skimming through the code I see mentions of: 
- dragging/dropping
- selecting
- node management with deletion, etc..
- key handling
- runtime events handling
all of which IMO could be considered as a separate responsibility.

I'm fine with you keeping the structure as it is if you don't find it more difficult to navigate and deal with the code.

It would be just interesting to understand what would be a good examples of:
- semantic grouping
- principles of single responsibility
So that it's easier to know when is the right time to consider splitting/extracting things out?

Reply:
hey Vytautas, I understand your points and both points of view on the subject.

there is logic here that could be separated in different files according to its different responsibilities, but we would have to decide on which way to go about that:
 - i've previously suggested a non-reusable hook, but that also has it's own disadvantages and might be a misleading pattern?
 - i could try to move much of this logic into functions in separate files, but there could be obstacles in passing all the arguments we need to those functions without making them bloated - however there might be a way to make it be clean enough with some effort

Reply:
in regards to the many nested `if` statements, i might at least be able to make things flatter by just stopping the function execution when some of them are false - might be more doable in some cases than other but i can always make an extra effort with that in mind

Reply:
> For now please disregard my suggestion and feel free to proceed with merging when you feel comfortable. I think we could have a separate discussion where we discuss these sort of things, until we have a clear path and agreement we should not block development :)

ok, i have it noted to try to improve and make this component more concise and readable whenever i have the chance - but if we can set some common practices and standards that might be helpful!